py_zipkin/zipkin.py

```diff
@@ -256,7 +256,8 @@ def __init__(
         if self.sample_rate is not None and not (0.0 <= self.sample_rate <= 100.0):
             raise ZipkinError('Sample rate must be between 0.0 and 100.0')
 
-        if not isinstance(self._span_storage, storage.SpanStorage):
+        if not (isinstance(self._span_storage, storage.SpanStorage) or
```

Comment:
That is another place that shows it would be nice to have a base type for the storage to support other implementations in this check and make the message more relevant.
Will it work if instance of `Stack` is passed?

Reply:
Yup, I'll change this.

> Will it work if instance of Stack is passed?

Nope, `Stack` is not iterable so it won't work.

Reply:
Sorry, my comment is misleading - stack is used separately from the storage and is not related here.